querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQuery.java

```diff
@@ -347,7 +348,8 @@ protected Configuration getConfiguration() {
         SQLListenerContextImpl context = startContext(connection(), queryMixin.getMetadata());
         String queryString = null;
         List<Object> constants = ImmutableList.of();
-
+        PreparedStatement stmt = null;
+        ResultSet rs = null;
         try {
```

Comment:
Please use `try-with-resources`.

Reply:
`try-with-resources` was not appropiate here, because in the happy flow the resources need to be closed lazily through the `SQLResultIterator`. The proper fix was to properly invoke `endContext` in the catch block for non-SQL runtime errors, as the context is responsible for freeing the resources in either scenario.